### PR TITLE
Fix carts rules without code

### PR DESCRIPTION
**Pull request by tomlev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | Fixes usage of empty code in cart rules : these cart rules don't need to be added manually to apply. Upgrade DB to deactivate cart rules with empty code (avoid side effects)
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-3307
| How to test?  | BO: create a cart rule with empty code. FO: check cart rule applies automatically. Check with restrictions too (carrier specific restriction, dates, ...)

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8809)

### CODE REVIEW

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8809#-:-L6q7I1Ma8Fj_FgfOgQ3:bnfp4nl)*

**Review by eternoendless — APPROVED** [review]

**Comment by marionf** [comment]
@tomlev 

If I have a restriction in my cart rule on the default carrier:

capture du 2018-03-07 11-57-31

capture du 2018-03-07 12-01-16

In the cart the promotion isn't displayed automatically on the shipping method step
capture du 2018-03-07 11-59-00

It appears if I select the other carrier and again the default one or on the payment step
capture du 2018-03-07 11-59-28

It should appear when I arrive on the shipping method step

File: classes/Cart.php [hunk 1]
```diff
@@ -470,6 +473,28 @@ public function getCartRules($filter = CartRule::FILTER_ACTION_ALL)
                 '.($filter == CartRule::FILTER_ACTION_REDUCTION ? 'AND (reduction_percent != 0 OR reduction_amount != 0)' : '')
                 .' ORDER by cr.priority ASC'
             );
+            $result = $activatedCartRulesRows;
```
Comment (Quetzacoalt91) [inline classes/Cart.php:476]: Why did you copy that variable? To avoid too many changes in the diff?

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by tomlev** [comment]
closed as duplicate of https://github.com/PrestaShop/PrestaShop/pull/8411

**Pull request closed without merging**